### PR TITLE
Te 55 create decryptor module

### PR DESCRIPTION
**Pull request by shriftz**
- Added docker related files to publish module to docker bub
- Folder structure changes

### CODE REVIEW

**Review by europaul — APPROVED** [review]

**Comment by europaul** [comment]
@unixdj could you review the go files here? you had many invaluable comments!

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
+
+[makefile]
+indent_style = tab
+indent_size = 4
```
Comment (unixdj) [inline .editorconfig:17]:
```suggestion
indent_size = 8
```

File: docker/docker-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
```
Comment (unixdj) [inline docker/docker-entrypoint.sh]: anything bash-specific in this script?

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,22 +1,15 @@
...
-# Copy the source code. Note the slash at the end, as explained in
-# https://docs.docker.com/engine/reference/builder/#copy
-COPY *.go ./
-COPY docker.env ./
-
-# Build
-RUN go build -o /serial-data-decrypter
+###
+FROM alpine:edge
+# Oh yes, you need bash!
```
Comment (unixdj) [inline docker/Dockerfile]: Yuck.  Bash.

File: docker/docker-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    : "${INGRESS_HOST:?Need to set INGRESS_HOST environment variable to string}"
+    : "${INGRESS_PORT:?Need to set INGRESS_PORT environment variable to string}"
+else
+    echo "Unrecognized MODULE_TYPE = $MODULE_TYPE, choose from Input, Processing, Output"
+    exit 1
+fi
+echo "[ENTRYPOINT] Environment validated."
+
+# CALL THE MAIN SCRIPT
+./app
```
Comment (unixdj) [inline docker/docker-entrypoint.sh]: Text files, especially shell scripts and source code, must end with a newline.

Guess how traditional `cpp` treated `#include` statements.

File: docker/docker-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+# Validate the environment according to module type
+if [[ "$MODULE_TYPE" == "Input" ]]
+then
+    : "${EGRESS_URLS:?Need to set EGRESS_URLS environment variable to string}"
+elif [[ "$MODULE_TYPE" == "Processing" ]]
+then
+    : "${INGRESS_HOST:?Need to set INGRESS_HOST environment variable to string}"
+    : "${INGRESS_PORT:?Need to set INGRESS_PORT environment variable to string}"
+    : "${EGRESS_URLS:?Need to set EGRESS_URLS environment variable to string}"
+elif [[ "$MODULE_TYPE" == "Output" ]]
```
Comment (unixdj) [inline docker/docker-entrypoint.sh]: normally spelled as `=`

File: example.env [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+MODULE_TYPE=Processing
+
+# REQUIRED API SETTINGS
+LOG_LEVEL=DEBUG
+INGRESS_HOST=0.0.0.0
+INGRESS_PORT=80
+EGRESS_URLS=https://hookb.in/example1,https://hookb.in/example2
+
+# YOU CAN ADD REMAINING MODULE SETTINGS BELOW
+AES_KEY=AAECAwQFBgcICQoLDA0ODw==
```
Comment (unixdj) [inline example.env:12]: Environment is not the best place for secrets.

File: makefile [hunk 7]
```diff
@@ -0,0 +1,18 @@
+MODULE=weevenetwork/serial-data-decryptor
+VERSION_NAME=v1.0.0
+
+create_image:
+	docker build -t ${MODULE}:${VERSION_NAME} . -f docker/Dockerfile
+.phony: create_image
```
Comment (unixdj) [inline makefile]: It's `.PHONY`, not `.phony`.  (Here and in other places.)

Does it actually work?  (You can check by creating a file `run_image`.)

File: makefile [hunk 8]
```diff
@@ -0,0 +1,18 @@
+MODULE=weevenetwork/serial-data-decryptor
```
Comment (unixdj) [inline makefile]: These files are traditionally named `Makefile`, capitalised.  Not a technical requirement, just conservatism.

File: src/utility/env.go [hunk 9]
```diff
@@ -6,9 +6,9 @@ import (
 )
 
 func GetEnvAsserted(envVarName string) string {
```
Comment (unixdj) [inline src/utility/env.go]: Panicking functions in Go are traditionally called `Must*`, like `regexp.MustCompile` and such.  May or may nor make sense for fatalling functions.

Comment (unixdj) [inline src/utility/env.go]: `envVarName`?  Either use `name` or something shorter, or go full microsoft hungarian notation and name it `sNameOfTheEnvironmentVariableThatWeAreTryingToRetrieveFromTheEnvironmentThatTherProcessIsRunningIn`.

From package "os":
```
func Getenv(key string) string
```

Also, there's this `Getenv` and there's the wonderfully named `godotenv` package (which doesn't parse `go.env`, of course, but it's ported from Ruby so of course it's idiotic).  I didn't look into the interaction between the two, but I hope you know what you're doing.

**Review by unixdj — DISMISSED** [review]

File: makefile (hunk 7 above)
Comment (europaul) [reply makefile]: you are right, it doesn't. nice catch!

File: src/utility/env.go (hunk 9 above)
Comment (europaul) [reply src/utility/env.go]: I don't really think that fits here

File: example.env (hunk 6 above)
Comment (europaul) [reply example.env:12]: this is gonna change as soon as we introduce the secrets management

**Pull request merged**